### ISSUE TITLE
Go to the waypoint specified in the url hash

### ISSUE THREAD

**Issue by vincentfretin** [c-frame/aframe-gltf-model-plus#42]
Go to the waypoint specified in the url hash like `#WPSit` instead of the spawn point, changes to be done in `move-to-spawn-point` to check the hash in the url, look at the `open-link` for the relevant part of getting the waypoint.

**Comment by uhunkler** [comment]
I can take this issue.

**Comment by vincentfretin** [comment]
#47 implements it. Thanks @uhunkler!

**Comment by vincentfretin** [comment]
I'm reopening, we implemented it partially. It works on the first load of the page only. That's already great and it answers my immediate need for it.
But we should also handle dynamically setting the hash in the url and listen to the [hashchange event](https://developer.mozilla.org/en-US/docs/Web/API/HashChangeEvent)
Test case, go to 
http://localhost:8080/examples/playground/index.html
then add the hash
http://localhost:8080/examples/playground/index.html#WPTwo
it should work, without refreshing the page.

**Comment by vincentfretin** [comment]
A side note, I believe hubs link component is working like this: it sets the hash in the url, go the waypoint and remove the hash in the url. I'm not sure we want this behavior for our open-link component, I think it's okay to keep the current implementation and keep the hash in url.

**Comment by jywarren** [comment]
Oof, I did notice that waypoints were exported from Spoke, but rooms and room contents from Hubs - linked by the .json property `"scene_id": "_________",` -- I am going to try writing a function to fetch the appropriate scene data and import waypoints, but just checking - this hasn't been done already or in an easier way? Thank you!!

**Comment by vincentfretin** [comment]
I have no idea what you're talking about. If you can get me an archive of one of your exported room, that would help, because I'm not aware of any json file.

**Comment by jywarren** [comment]
No worries, I can figure it out. But in case others are working through this, each Hubs room (from the standard Hubs backup tool) in contains a .json file (`/rooms/<room_id>/<room_id>.json` ) with the unique ID of the room (from the URL). That .json file references a scene ID. Unfortunately the scenes are in `/spoke/<some_id>/` folders, and the folder name does NOT match the scene ID. I had to manually search the folders until I found the scene ID. But the scene ID DOES match the name of the .json file in the scene folder. (`/spoke/<project_id>/<scene_id>.json`)

That `<scene_id>.json` file then references the project URL:

`"project_url": "https://uploads-prod.reticulum.io/files/6b9f476d-ff64-4423-bd5c-420e74acd18a.json",`

which can be used to find the 3rd .json file, which is actually a local file of that same name, so in this example `6b9f476d-ff64-4423-bd5c-420e74acd18a.json`

That file contains waypoint data. I've included examples of all 3 files here for reference:

https://gist.github.com/jywarren/f92ecc481119a05a1336046bf3307dba

Whew, that's a pain! Will try to report back if I figure out how the project_id is determined starting from the room export.

**Comment by jywarren** [comment]
I think I'll skip trying to automate locating the file, and instead manually associate it, and try to parse the JSON file to generate waypoint a-frame entities or something at the given location/orientation, with the correct ids.

**Comment by jywarren** [comment]
Well, objects.gltf only contains user-added items. The waypoints are added at the scene/spoke level. So I understand why they are stored in a different folder, I just wish the export designers left a path reference for where the scene data could be easily found, if you start from a room export. 

But yes, also, room metadata specific to Hubs is stored in a json file, not in objects.gltf.

**Comment by vincentfretin** [comment]
Spoke project may be in json, but the .bin should contains all the waypoints as gltf extensions already no?

**Comment by jywarren** [comment]
I guess not - maybe the issue is that when downloading a room from the console as I did in https://github.com/c-frame/aframe-gltf-model-plus/issues/18, you get a .bin file which is actually a .glb - but the Hubs Exporter does not generate that file. It generates the ones I saved a sample of. It's too bad! 

I may be missing something but that's my best understanding right now.

**Comment by vincentfretin** [comment]
But I see
`    "model_url": "https://uploads-prod.reticulum.io/files/2b401e8b-bee5-4c74-af8b-9b2366cc06f3.bin",`
in your json. You don't have the file?

**Comment by jywarren** [comment]
That's actually the .json for the scene, not the room. The file does exist in the scene export folder, though, in both .glb and .bin formats. I guess waypoints are provided in all three, but indeed, I can probably include that file and get all the waypoints quickly. 

Auto-detecting the scene would be even nicer, but this is a good start, let me try it.

**Comment by jywarren** [comment]
Here is the .glb from the scene: http://archive.publiclab.org/scene.glb

I loaded it but the waypoints didn't seem to work via the URL hash. But I'm not sure how to inspect a .glb in depth in that way - i did open it in https://sandbox.babylonjs.com/ but not sure that can detect waypoints. Any advice on how I should look into the file to check? Thank you very much.

**Comment by vincentfretin** [comment]
All the gltf extensions are logged in the Chrome console, I see all your images and other components in the logs, but I don't see any waypoints.

**Comment by jywarren** [comment]
OK, I wrote a small parser. I didn't look into the correct timings for when to run it so I just delayed by ~100ms. It works for me, and for what it's worth, parts could be re-used by others looking to access Hubs exported waypoints. I just fetched the scene JSON manually and scanned it for waypoints, and manually scanned the URL hash, then matched the two whenever it changes:

https://gist.github.com/jywarren/e5b9b1812064497328a84aafdcb58bcf
